plugin/pluginenv/environment.go

```diff
@@ -35,19 +45,55 @@ func New(options ...Option) (*Environment, error) {
 	}
 	if env.searchPath == "" {
 		return nil, fmt.Errorf("a search path must be provided")
-	} else if env.apiProvider == nil {
-		return nil, fmt.Errorf("an api provider must be provided")
 	}
 	return env, nil
 }
 
+// Returns the configured webapp path.
+func (env *Environment) WebappPath() string {
+	return env.webappPath
+}
+
+// Returns the configured search path.
+func (env *Environment) SearchPath() string {
+	return env.searchPath
+}
+
 // Returns a list of all plugins found within the environment.
-func (env *Environment) Plugins() ([]*plugin.BundleInfo, error) {
+func (env *Environment) Plugins() ([]*model.BundleInfo, error) {
+	env.mutex.Lock()
+	defer env.mutex.Unlock()
 	return ScanSearchPath(env.searchPath)
 }
 
+// Returns a list of all currently active plugins within the environment.
+func (env *Environment) ActivePlugins() ([]*model.BundleInfo, error) {
+	env.mutex.Lock()
+	defer env.mutex.Unlock()
+
+	plugins, err := ScanSearchPath(env.searchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	activePlugins := []*model.BundleInfo{}
+	for id := range env.activePlugins {
+		for _, p := range plugins {
+			if p.Manifest != nil && p.Manifest.Id == id {
+				activePlugins = append(activePlugins, p)
+				break
+			}
+		}
+	}
```

Comment:
Why are you doing all this instead of...

```go
for id, activePlugin := range env.activePlugins {
    activePlugins = append(activePlugins, activePlugin.BundleInfo)
}
```

?

That would be more guaranteed to return the bundle path and manifest version that's actually active.

Reply:
Sigh, yeah that was a goof. Fixed